express-error-handling/server/server.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { readFile, writeFile } from 'node:fs/promises';
+import express from 'express';
+import ClientError from './client-error.js';
+import errorMiddleware from './error-middleware.js';
+
+const app = express();
+app.use(express.json());
+
+async function readData() {
+  const data = await readFile('./data.json');
+  return JSON.parse(data);
+}
+
+async function writeData(data) {
+  await writeFile('./data.json', JSON.stringify(data, null, 2));
+}
+
+app.get('/api/notes', async (req, res, next) => {
+  try {
+    const data = await readData();
+    const notes = [];
+    for (const id in data.notes) {
+      notes.push(data.notes[id]);
+    }
+    res.json(notes);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res, next) => {
+  try {
+    const data = await readData();
+    const id = Number(req.params.id);
+    if (Number.isNaN(id) || !Number.isInteger(id) || id < 1) {
+      try {
```

Comment:
No, this isn't what you want. You don't need a try/catch around the `throw`. You should let the single try/catch around the entire function catch the exception.